src/lib-http/fuzz-http-url.c

```diff
@@ -0,0 +1,35 @@
+
+#include <stdint.h>
+
+#include "lib.h"
+#include "net.h"
+#include "http-url.h"
+#include "test-common.h"
+
+
+int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size);
+int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    T_BEGIN{
+    struct http_url *urlp = NULL;
+    const char *error = NULL;
+
+    // Allocate string we null terminate
+	char *new_data = malloc(size+1);
+    memcpy(new_data, data, size);
+	if (new_data == NULL)
+        return 0;    
+    new_data[size] = '\0';
```

Comment:
All this could be done more easily with `char *new_data = i_strndup(data, size); ... i_free(new_data);` or since you have the T_BEGIN/T_END anyway could also be `const char *new_data = t_strndup(data, size);`